### PR TITLE
RFC6265bis: Use navigables concept

### LINKED ISSUE #2372: Fix browsing context integration in 6265bis
6265bis has a bunch of references to browsing context and its active document. There were some big changes to HTML recently  in https://github.com/whatwg/html/pull/6315 that should have broken some of those.

Note that a lot of this could be solved through adding more accurate layering as described in #2084, but if we wanted to publish a new cookies RFC before that change happens, it may still be worth adding the correct terminology and references. 

cc @annevk @domenic @sbingler

### PR DESCRIPTION
**Pull request by sbingler**
Closes #2372

The HTML spec recommends avoiding using "browser contexts" and instead recommends "navigables".

This PR refactors rfc6265bis to use "navigables".

### CODE REVIEW

**Comment by sbingler** [comment]
@johannhof PTAL

@jakearchibald If you have any time. These concepts look like they should be drop in replacements but I'd appreciate your feedback just in case.

File: draft-ietf-httpbis-rfc6265bis.md
Comment (sbingler) [inline draft-ietf-httpbis-rfc6265bis.md:1715]: `target browsing context` is defined by FETCH and it doesn't look like it's been changed (yet?)

File: draft-ietf-httpbis-rfc6265bis.md
Comment (jakearchibald) [inline draft-ietf-httpbis-rfc6265bis.md:1103]: Similar to above, I'd reword this to avoid assuming that a document is its navigable's active document. Or, if you're sure that this algorithm only runs when the chain of documents are active, add a line that asserts that.

**Review by jakearchibald — APPROVED** [review]
Couple of nits, but LGTM

**Comment by johannhof** [comment]
Thanks @sbingler, this looks conceptually right to me but I'll defer to @jakearchibald for detailed inspection :)

File: draft-ietf-httpbis-rfc6265bis.md
Comment (domenic) [reply draft-ietf-httpbis-rfc6265bis.md:1719]: This is https://github.com/whatwg/html/issues/8463 . In the meantime you can use "target browsing context's navigable is a top-level traversable".

File: draft-ietf-httpbis-rfc6265bis.md
Comment (domenic) [reply draft-ietf-httpbis-rfc6265bis.md:1104]: I think you want to use https://html.spec.whatwg.org/#inclusive-ancestor-navigables . See e.g. this call site for something that does exactly what you're doing: https://html.spec.whatwg.org/#the-autofocus-attribute:inclusive-ancestor-navigables

You kind of need to care about activeness because you need a non-ambiguous way of going from navigables (which are in a tree) to documents (which are not). Each navigable has a single active document and potentially many inactive documents; the latter are stored in bfcache. You want to pick out the active document.

E.g. given:

```
parent navigable: A --> [B] --> C
                         |
child navigable:      [child]
```

where `[]` denotes the currently-active document, you want the set to contain B, but not A or C.

Comment (sbingler) [reply draft-ietf-httpbis-rfc6265bis.md]: Thanks for all the feedback, I used the inclusive ancestor navigables concept. I also kept the note that all docs should be active, @mikewest do you have a preference to remove/keep it?

File: draft-ietf-httpbis-rfc6265bis.md
Comment (sbingler) [inline draft-ietf-httpbis-rfc6265bis.md:1900]: @domenic Can I get your thoughts on this bullet point?

Comment (domenic) [reply draft-ietf-httpbis-rfc6265bis.md:1900]: I guess the most proper way would be "is the [active browsing context](https://html.spec.whatwg.org/#nav-bc) or a top-level traversable". But hopefully we will fix https://github.com/whatwg/html/issues/8463 at some point soon so you don't need these circumlocutions...

Comment (sbingler) [reply draft-ietf-httpbis-rfc6265bis.md:1900]: Thanks,

Is there an ETA for that issue? I'm fine with holding off on this change until that lands

Comment (domenic) [reply draft-ietf-httpbis-rfc6265bis.md:1900]: No specific ETA, sorry.

**Comment by sbingler** [comment]
@mikewest ping

**Review by mikewest — APPROVED** [review]
Yup. LGTM.

Thanks!

**Comment by sbingler** [comment]
Thanks Mike, conflicts should be resolved

**Pull request merged**
